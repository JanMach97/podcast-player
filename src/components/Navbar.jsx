import React from 'react';

const Navbar = () => {
  return (
    <div>
      <h1>Podcast Nav</h1>
    </div>
  );
};

export default Navbar;