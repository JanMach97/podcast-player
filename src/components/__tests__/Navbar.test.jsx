import * as React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Navbar from "../Navbar";

configure({ adapter: new Adapter() });

describe("Navbar", () => {
  it("renders without crashing", () => {
    const component = shallow(<Navbar />);
    expect(component).toMatchSnapshot();
  });
  it("logo links to main page", () => {
    const component = shallow(<Navbar />);
    expect(component.find("Link").prop("to")).toBe("/");
  });
});
