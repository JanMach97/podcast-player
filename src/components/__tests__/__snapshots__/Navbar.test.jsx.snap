// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
        >
          <div
            className="logo"
          >
            <i
              className="fas fa-microphone"
            />
            <h1>
              Podcast
            </h1>
          </div>
        </Link>,
        <input
          onChange={[Function]}
          placeholder="Search Episode"
          type="text"
          value=""
        />,
      ],
      "className": "navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="logo"
          >
            <i
              className="fas fa-microphone"
            />
            <h1>
              Podcast
            </h1>
          </div>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-microphone"
              />,
              <h1>
                Podcast
              </h1>,
            ],
            "className": "logo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-microphone",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Podcast",
              },
              "ref": null,
              "rendered": "Podcast",
              "type": "h1",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Search Episode",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
          >
            <div
              className="logo"
            >
              <i
                className="fas fa-microphone"
              />
              <h1>
                Podcast
              </h1>
            </div>
          </Link>,
          <input
            onChange={[Function]}
            placeholder="Search Episode"
            type="text"
            value=""
          />,
        ],
        "className": "navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="logo"
            >
              <i
                className="fas fa-microphone"
              />
              <h1>
                Podcast
              </h1>
            </div>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-microphone"
                />,
                <h1>
                  Podcast
                </h1>,
              ],
              "className": "logo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-microphone",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Podcast",
                },
                "ref": null,
                "rendered": "Podcast",
                "type": "h1",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Search Episode",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
